Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -632,9 +630,9 @@ String ImageDecoderCG::decodeUTI(CGImageSourceRef imageSource, const SharedBuffe
 
     auto boxUnsigned = [span = data.span()](unsigned index) -> unsigned {
         constexpr bool isLittleEndian = false;
-        const unsigned* boxBytes = reinterpret_cast<const unsigned*>(span.data());
+        auto value = reinterpretCastSpanStartTo<const unsigned>(span.subspan(index * sizeof(unsigned)));
         // Numbers in the file are BigEndian.
-        return flipBytesIfLittleEndian(boxBytes[index], isLittleEndian);
+        return flipBytesIfLittleEndian(value, isLittleEndian);
```

Comment:
Not new to this patch: This a strange way to write this, the way it involves a named boolean argument. I wish we used a more straightforward idiom.

For example, if I was writing this I would read 4 bytes out as a big-endian value, which we must do in at least some places in WebKit. That would be something like `readBigEndian<uint32_t>` function that takes a `std::span<uint8_t>` argument, avoiding both the `reinterpret_cast` and the "flip bytes" concept.

But if we are going to do it like this, with "flipping", the `flipBytesIfLittleEndian` function could have an overload that doesn’t take a boolean argument. Given the function name it seems logical that it would flip the bytes in the passed-in value iff the platform’s byte ordering is little-endian.